includes/REST/OrderControllerV2.php

```diff
@@ -0,0 +1,288 @@
+                            'description' => __( 'Download IDs.', 'dokan-lite' ),
+                            'required'    => true,
+                            'items' => [
+                                'type'     => 'integer',
+                                'description' => __( 'Download product IDs.', 'dokan-lite' ),
+                                'required' => true,
+                            ],
+                        ],
+                    ],
+                ],
+                [
+                    'methods'             => WP_REST_Server::DELETABLE,
+                    'callback'            => [ $this, 'revoke_order_downloads' ],
+                    'permission_callback' => [ $this, 'get_single_order_permissions_check' ],
+                    'args'                => [
+                        'download_id' => [
+                            'type'        => 'string',
+                            'description' => __( 'Download ID.', 'dokan-lite' ),
+                            'required'    => false,
+                        ],
+                        'product_id' => [
+                            'type'        => 'integer',
+                            'description' => __( 'Product ID.', 'dokan-lite' ),
+                            'required'    => false,
+                        ],
+                        'permission_id' => [
+                            'type'        => 'integer',
+                            'description' => __( 'Permission ID.', 'dokan-lite' ),
+                            'required'    => true,
+                        ],
+                    ],
+                ],
+            ]
+        );
+
+        register_rest_route(
+            $this->namespace, '/' . $this->base . '/bulk-actions', [
+				[
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'process_orders_bulk_action' ),
+					'args'                => array(
+						'order_ids' => array(
+							'type'        => 'array',
+							'description' => __( 'Order ids', 'dokan-lite' ),
+							'required'    => true,
+							'sanitize_callback' => [ $this, 'sanitize_order_ids' ],
+						),
+						'status' => array(
+							'type'        => 'string',
+							'description' => __( 'Order status', 'dokan-lite' ),
+							'required'    => true,
+							'sanitize_callback' => 'sanitize_text_field',
+						),
+					),
+					'permission_callback' => array( $this, 'update_order_permissions_check' ),
+                ],
+            ]
+        );
+    }
+
+    /**
+     * Get Order Downloads.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @param  \WP_REST_Request $requests Request object.
+     *
+     * @return WP_Error|\WP_HTTP_Response|\WP_REST_Response
+     */
+    public function get_order_downloads( $requests ) {
+        global $wpdb;
+
+        $user_id   = dokan_get_current_user_id();
+        $data      = [];
+        $downloads = [];
+
+        $download_permissions = $wpdb->get_results(
```

Comment:
kindly add a todo here that we need to move this bit of code to a separate function.